components/molecules/Pagination/pagination.tsx

```diff
@@ -43,42 +45,63 @@ const Pagination = ({
+      ? index + 1
+      : selected >= pages.length - Math.floor(pageSize / 2)
+      ? pages.length - pageSize + index + 1
+      : selected - Math.floor(pageSize / 2) + index
+  ).filter((page) => page >= 1 && page <= pages.length);
+
   return (
     <>
       <div className="w-max flex gap-x-4 items-center">
         <div className="flex items-center gap-x-4">
           <button
             className="text-light-slate-9 disabled:text-light-slate-7"
-            disabled={!hasPreviousPage ? true : false}
-            onClick={() => handlePrev()}
+            disabled={!hasPreviousPage}
+            onClick={() => {
+              if (!hasPreviousPage) {
+                return;
+              }
+
+              handlePrev();
+              setSelected(selected === 1 ? 1 : selected - 1);
+            }}
           >
-            <RiArrowLeftSLine onClick={() => handlePrev()} className="text-2xl md:text-lg" />
+            <RiArrowLeftSLine className="text-2xl md:text-lg" />
           </button>
-          {pages.map((page, index) => {
+          {middlePages.map((page) => {
             return (
-              index < pageSize && (
-                <button
-                  key={index}
-                  onClick={() => {
-                    setSelected(page);
-                    handleSelected(page);
-                  }}
-                  className={`${
-                    // this check  will be updated from page to currentPage when the implemetation of logic is ready
-                    selected === page &&
-                    "border !text-light-slate-12 shadow-paginate border-light-orange-10 bg-light-orange-2 shadow-search"
-                  } cursor-pointer text-light-slate-11 transition text-sm rounded-lg w-8 h-8 hover:bg-light-orange-2 hover:text-light-orange-10`}
-                >
-                  {page}
-                </button>
-              )
+              <button
+                key={page}
+                onClick={() => {
+                  setSelected(page);
+                  handleSelected(page);
+                }}
+                className={`${
+                  // this check  will be updated from page to currentPage when the implemetation of logic is ready
+                  selected === page &&
+                  "border !text-light-slate-12 shadow-paginate border-light-orange-10 bg-light-orange-2 shadow-search"
+                } cursor-pointer text-light-slate-11 transition text-sm rounded-lg w-8 h-8 hover:bg-light-orange-2 hover:text-light-orange-10`}
+              >
+                {page}
+              </button>
             );
           })}
 
           <button
             className="text-light-slate-9 disabled:text-light-slate-7"
             disabled={!hasNextPage ? true : false}
-            onClick={() => handleNext()}
+            onClick={() => {
+              if (!hasNextPage) {
+                return;
+              }
+
+              handleNext();
+              debugger;
```

Comment:
```suggestion
```

Reply:
😬 We should a lint rule for that.

Reply:
I created #2082 for that.

Reply:
Tnx 💯

Reply:
PR is up for the `no-debugger` ESLint rule. 😅 #2083